elyra/pipeline/validation.py

```diff
@@ -422,23 +419,38 @@ async def _validate_custom_component_node_properties(self,
                     # Any component property with type `InputPath` will be a dictionary of two keys
                     # "value": the node ID of the parent node containing the output
                     # "option": the name of the key (which is an output) of the above referenced node
-                    if not isinstance(node_param_value, dict) or \
-                            len(node_param_value) != 2 or \
-                            set(node_param_value.keys()) != {'value', 'option'}:
+                    if not isinstance(node_param, dict) or len(node_param) != 2 or \
+                            set(node_param.keys()) != {'value', 'option'}:
                         response.add_message(severity=ValidationSeverity.Error,
                                              message_type="invalidNodeProperty",
                                              message="Node has malformed `InputPath` parameter structure",
                                              data={"nodeID": node.id,
                                                    "nodeName": node.label})
                     node_ids = list(x.get('node_id_ref', None) for x in node.component_links)
                     parent_list = self._get_parent_id_list(pipeline_definition, node_ids, [])
-                    if node_param_value.get('value') not in parent_list:
+                    node_param_value = node_param.get('value')
+                    if node_param_value not in parent_list:
                         response.add_message(severity=ValidationSeverity.Error,
                                              message_type="invalidNodeProperty",
                                              message="Node contains an invalid inputpath reference. Please "
                                                      "check your node-to-node connections",
                                              data={"nodeID": node.id,
                                                    "nodeName": node.label})
+                elif isinstance(node_param, dict) and node_param.get('activeControl') == 'NestedEnumControl':
+                    # TODO: Update this hardcoded check for xcom_push. This parameter is specific to a runtime
```

Comment:
The code below appears to assume that `NestedEnumControl` is only applicable to Airflow because line 446 (`if not xcom_value:`) will trigger an error if no `xcom_value` is present. For runtime types other than Airflow, lines 443 onwards would all yield unexpected results. Instead of deferring a resolution, the code below should already tolerate the absence of `xcom_*` in the pipeline file.

Reply:
I can push a basic fix for this (i.e. just check for the existence of the `xcom_param` before moving on). But it looks like we might still want something more robust for a long-term solution

Reply:
Yes.